### PR TITLE
EMO-6419: Added ability to set batch size thresholds

### PR DESCRIPTION
**Pull request by billkalter**

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/defaultimpl/StandardTuner.java [hunk 1]
```diff
@@ -165,6 +169,16 @@ private void configureCommitLogBackups() throws IOException {
         }
     }
 
+    private void configureBatchSizes(CassandraConfiguration cassandraConfiguration, Map<String, Object> yaml, @Nullable VersionNumber cassandraVersion) {
+        put(yaml, "batch_size_warn_threshold_in_kb", cassandraConfiguration.getBatchSizeFailureThresholdInKb());
```
Comment (sujithvaddi) [inline priam/src/main/java/com/netflix/priam/defaultimpl/StandardTuner.java]: Even though the cassandraVersion is nullable, should we throw a log warning or something(not exactly at this line) if it's null, as ideally cassandra -v should throw the version output and our parsing has to get that value if there no unknown errors.

**Comment by billkalter** [comment]
@sujithvaddi Addressed feedback

File: priam/src/main/java/com/netflix/priam/defaultimpl/StandardTuner.java [hunk 2]
```diff
@@ -170,12 +170,22 @@ private void configureCommitLogBackups() throws IOException {
...
+        if (batchSizeFailureThresholdInKb != null) {
+            if (cassandraVersion == null) {
+                logger.warn("Batch size failure threshold has been set to {} but the Cassandra version could not be confirmed. " +
+                        "If Cassandra version is not 2.2+ this will cause a configuration error.", batchSizeFailureThresholdInKb);
+            }
+            if (cassandraVersion != null && cassandraVersion.compareTo(VersionNumber.parse("2.2")) < 0) {
+                logger.info("Batch size failure threshold has been set to {} but Cassandra version {} does not support this" +
+                        "option.  Ignoring value.", batchSizeFailureThresholdInKb, cassandraVersion);
+            } else {
+                put(yaml, "batch_size_fail_threshold_in_kb", cassandraConfiguration.getBatchSizeFailureThresholdInKb());
```
Comment (sujithvaddi) [inline priam/src/main/java/com/netflix/priam/defaultimpl/StandardTuner.java]: Minor: why not just use batchSizeFailureThresholdInKb in the put as well.

Comment (billkalter) [reply priam/src/main/java/com/netflix/priam/defaultimpl/StandardTuner.java]: Just missed it, I'll update

**Pull request merged**